extensions/replication/ReplicationConfigValidator.js

```diff
@@ -103,21 +145,32 @@ function _loadAdminCredentialsFromFile(filePath) {
 function configValidator(backbeatConfig, extConfig) {
     const validatedConfig = joi.attempt(extConfig, joiSchema);
     const { source, destination } = validatedConfig;
-
     if (source.auth.vault) {
         const { adminCredentialsFile } = source.auth.vault;
         if (adminCredentialsFile) {
             source.auth.vault.adminCredentials =
                 _loadAdminCredentialsFromFile(adminCredentialsFile);
         }
     }
-    if (destination.auth.vault) {
-        const { adminCredentialsFile } = destination.auth.vault;
+    if (destination.auth?.vault) {
+        const { adminCredentialsFile } = source.auth.vault;
         if (adminCredentialsFile) {
             destination.auth.vault.adminCredentials =
                 _loadAdminCredentialsFromFile(adminCredentialsFile);
         }
     }
+    if (destination.sites) {
+        Object.values(destination.sites).forEach(site => {
+            if (site.auth.vault) {
+                const { adminCredentialsFile } = site.auth.vault;
+                if (adminCredentialsFile) {
+                    // eslint-disable-next-line no-param-reassign
+                    site.auth.vault.adminCredentials =
+                        _loadAdminCredentialsFromFile(adminCredentialsFile);
+                }
+            }
```

Comment:
set the default vault already, so that "config" handling is done only here (and does not trickle in the "business" code) : i.e. from here on, auth is always specified in each destination' site?

```suggestion
            } else {
                site.auth.vault = destination.auth.vault;
            }
```

Reply:
There is no guarantee we have all the sites here, so we would still have to check later. We could create and append the config of each site if it doesn't exist to the structure but that would just be redundant information, especially since we already go through the config class to get the destination site's config so it's already abstracted from the business logic.